### ISSUE TITLE
API to check if the wallet name available 

### ISSUE THREAD

**Issue by dadiorchen** [Greenstand/treetracker-wallet-api#343]
On the client/UI we don't want to get the duplicated wallet name till we do the creation, we want to check it once user types the name,so we need a API to check if a wallet name is available, including: the name format doesn't violate our name rules, the name isn't used by another wallet.

@Kpoke does this make sense?

**Comment by Kpoke** [comment]
Yeah, this definitely makes sense, do we want this in v1 or v2 or both?

**Comment by dadiorchen** [comment]
V2 is fine

**Comment by Kpoke** [comment]
Work should be done off the v1.10 branch

**Comment by Kpoke** [comment]
#364
